scripts/submission/submit_to_gencc.py

```diff
@@ -0,0 +1,505 @@
+    data: list, outfile: str, dry: str, db_config: dict[str, Any]
+) -> str:
+    """Creates the G2P_GenCC.txt and also calls the create the gencc_submission function when dry is False,
+    A real run
+    Also writes out records with no disease id
+
+    Args:
+        data (dict[str, Any]): Record of unsubmitted ids
+        outfile (str): Txt file to be created
+        dry (str): To allow for a real run, which updates the db
+        db_config (dict[str, Any]): DB/API configuration dictionary
+
+    Returns:
+        str: An output file
+    """
+    with open(outfile, mode="w") as output_file:
+        output_file.write(
+            "submission_id\thgnc_id\thgnc_symbol\tdisease_id\tdisease_name\tmoi_id\tmoi_name\tsubmitter_id\tsubmitter_name\tclassification_id\tclassification_name\tdate\tpublic_report_url\tpmids\tassertion_criteria_url\n"
+        )
+        issues_with_record = []
+        gencc_list = []
+        submission_id_base = "1000112"
+        for record in data:
+            g2p_id = record["g2p id"]
+            submission_id = submission_id_base + str(g2p_id[3:])
+
+            hgnc_id = record["hgnc id"]
+            hgnc_symbol = record["gene symbol"]
+
+            disease_id = record["disease mim"] or record["disease MONDO"]
+            if disease_id is None or disease_id == "":
+                issues_with_record.append(g2p_id)
+                continue
+            disease_name = record["disease name"]
+            moi_id = allelic_requirement[record["allelic requirement"]]
+            moi_name = record["allelic requirement"]
+            submitter_id = "GENCC:000113"
+            submitter_name = "G2P"
+            classification_id = confidence_category[record["confidence"]]
+            classification_name = record["confidence"]
+            dt = datetime.fromisoformat(record["date of last review"])
+            date = dt.strftime("%Y/%m/%d")
+            record_url = "https://www.ebi.ac.uk/gene2phenotype/lgd/" + g2p_id
+            pmids = record["publications"]
+            assertion_criteria_url = "https://www.ebi.ac.uk/gene2phenotype/terminology"
+
+            line_to_output = f"{submission_id}\t{hgnc_id}\t{hgnc_symbol}\t{disease_id}\t{disease_name}\t{moi_id}\t{moi_name}\t{submitter_id}\t{submitter_name}\t{classification_id}\t{classification_name}\t{date}\t{record_url}\t{pmids}\t{assertion_criteria_url}\n"
+            output_file.write(line_to_output)
+            if dry is True:
+                type_of = "create"
+                db_date = create_datetime_now()
+                created_record = create_gencc_submission_record(
+                    submission_id, db_date, type_of, g2p_id
+                )
+                gencc_list.append(created_record)
+            if len(issues_with_record) > 0:
+                with open("record_with_issues.txt", mode="w") as textfile:
+                    for issues in issues_with_record:
+                        textfile.write(f"{issues}\n")
```

Comment:
This is file is being opened N number of records because it is inside the loop `for record in data`